### ISSUE TITLE
Fix golint and broken mac tests

### ISSUE THREAD

**Issue by sjezewski** [pachyderm/pachyderm#456]
We've accumulated some golint errors:

```shell
src/server/pachyderm_test.go:1281: no formatting directive in Sprintf call
src/server/pachyderm_test.go:1281: no formatting directive in Sprintf call
src/server/pfs/drive/driver.go:824: missing argument for Errorf("%d"): format reads arg 1, have only 0 args
src/server/pfs/drive/driver.go:824: missing argument for Errorf("%d"): format reads arg 1, have only 0 args
src/server/pps/server/api_server.go:181: ppsclient.Job composite literal uses unkeyed fields
src/server/pps/server/api_server.go:931: persist.Shard composite literal uses unkeyed fields
src/server/pps/server/api_server.go:181: ppsclient.Job composite literal uses unkeyed fields
src/server/pps/server/api_server.go:931: persist.Shard composite literal uses unkeyed fields
```

And if you run `make test-fuse` on mac, we get some errors (relics from the client refactor):

```shell
CGOENABLED=0 GO15VENDOREXPERIMENT=1 go test -cover $(go list ./src/server/... | grep -v '/src/server/vendor/' | grep '/src/server/pfs/fuse')
# github.com/pachyderm/pachyderm/src/server/pfs/fuse_test
src/server/pfs/fuse/filesystem_seek_mac_test.go:27: cannot use func literal (type func("github.com/pachyderm/pachyderm/src/server/vendor/github.com/pachyderm/pachyderm/src/client/pfs".APIClient, string)) as type func(client.APIClient, string) in argument to testFuse
src/server/pfs/fuse/filesystem_seek_mac_test.go:29: undefined: "github.com/pachyderm/pachyderm/src/server/vendor/github.com/pachyderm/pachyderm/src/client/pfs".CreateRepo
src/server/pfs/fuse/filesystem_seek_mac_test.go:30: undefined: "github.com/pachyderm/pachyderm/src/server/vendor/github.com/pachyderm/pachyderm/src/client/pfs".StartCommit
src/server/pfs/fuse/filesystem_seek_mac_test.go:39: undefined: "github.com/pachyderm/pachyderm/src/server/vendor/github.com/pachyderm/pachyderm/src/client/pfs".FinishCommit
src/server/pfs/fuse/filesystem_seek_mac_test.go:84: cannot use func literal (type func("github.com/pachyderm/pachyderm/src/server/vendor/github.com/pachyderm/pachyderm/src/client/pfs".APIClient, string)) as type func(client.APIClient, string) in argument to testFuse
src/server/pfs/fuse/filesystem_seek_mac_test.go:86: undefined: "github.com/pachyderm/pachyderm/src/server/vendor/github.com/pachyderm/pachyderm/src/client/pfs".CreateRepo
src/server/pfs/fuse/filesystem_seek_mac_test.go:87: undefined: "github.com/pachyderm/pachyderm/src/server/vendor/github.com/pachyderm/pachyderm/src/client/pfs".StartCommit
src/server/pfs/fuse/filesystem_seek_mac_test.go:123: undefined: "github.com/pachyderm/pachyderm/src/server/vendor/github.com/pachyderm/pachyderm/src/client/pfs".FinishCommit
src/server/pfs/fuse/filesystem_seek_mac_test.go:134: undefined: "github.com/pachyderm/pachyderm/src/server/vendor/github.com/pachyderm/pachyderm/src/client/pfs".CreateRepo
src/server/pfs/fuse/filesystem_seek_mac_test.go:135: undefined: "github.com/pachyderm/pachyderm/src/server/vendor/github.com/pachyderm/pachyderm/src/client/pfs".StartCommit
```

Little things, but things we should clean up.

**Comment by jdoliner** [comment]
As part of this can we make it so that golint errors fail CI when they exist?
And although we can't run the mac test we could probably compile it in CI.

**Comment by sjezewski** [comment]
Good ideas. Yea, we should do those things. That makes these non repeatable.

Also want to add -- I want to remove the debug lines from `filesystem_test.go` -- I was hoping that would help us catch other intermittent fuse errors ... but I don't think thats the case.

**Comment by sambooo** [comment]
Tests fixed in #469 

Hugo checks for linting errors [in its Makefile](https://github.com/spf13/hugo/blob/master/Makefile#L54) in a way that will fail CI.

**Comment by sjezewski** [comment]
Great! Thanks!

Ok, so remaining todos are:

- removing debug lines from `testFuse()` helper
- updating the `filesystem_seek_mac_test.go` to compile

**Comment by sambooo** [comment]
Compilation of that file was fixed in #469, no changes made to testFuse() though

**Comment by jdoliner** [comment]
Looks like `testFuse` was cleaned up at some point. I don't see any debug lines left in it.
Pretty sure this issue is complete, closing.
